frontend/src/pages/CompareV2.test.tsx

```diff
@@ -936,4 +940,84 @@ describe('CompareV2', () => {
       screen.getByText('Error: failed loading HTML file. Click Details for more information.');
     });
   });
+
+  describe('EnhancedCompareV2', () => {
+    it('Redirects to experiments page when namespace changes', () => {
+      const history = createMemoryHistory({
+        initialEntries: ['/does-not-matter'],
+      });
+      const { rerender } = render(
+        <CommonTestWrapper>
+          <Router history={history}>
+            <NamespaceContext.Provider value='ns1'>
+              <EnhancedCompareV2 {...generateProps()} />
+            </NamespaceContext.Provider>
+          </Router>
+        </CommonTestWrapper>,
+      );
+      expect(history.location.pathname).not.toEqual('/experiments');
+      rerender(
+        <CommonTestWrapper>
+          <Router history={history}>
+            <NamespaceContext.Provider value='ns2'>
+              <EnhancedCompareV2 {...generateProps()} />
+            </NamespaceContext.Provider>
+          </Router>
+        </CommonTestWrapper>,
+      );
+      expect(history.location.pathname).toEqual('/experiments');
+    });
+
+    it('Does not redirect when namespace stays the same', () => {
+      const history = createMemoryHistory({
+        initialEntries: ['/initial-path'],
+      });
+      const { rerender } = render(
+        <CommonTestWrapper>
+          <Router history={history}>
+            <NamespaceContext.Provider value='ns1'>
+              <EnhancedCompareV2 {...generateProps()} />
+            </NamespaceContext.Provider>
+          </Router>
+        </CommonTestWrapper>,
+      );
+      expect(history.location.pathname).toEqual('/initial-path');
+      rerender(
+        <CommonTestWrapper>
+          <Router history={history}>
+            <NamespaceContext.Provider value='ns1'>
+              <EnhancedCompareV2 {...generateProps()} />
+            </NamespaceContext.Provider>
+          </Router>
+        </CommonTestWrapper>,
+      );
+      expect(history.location.pathname).toEqual('/initial-path');
+    });
+
+    it('Does not redirect when namespace initializes', () => {
+      const history = createMemoryHistory({
+        initialEntries: ['/initial-path'],
+      });
+      const { rerender } = render(
+        <CommonTestWrapper>
+          <Router history={history}>
+            <NamespaceContext.Provider value={undefined}>
+              <EnhancedCompareV2 {...generateProps()} />
+            </NamespaceContext.Provider>
+          </Router>
+        </CommonTestWrapper>,
+      );
+      expect(history.location.pathname).toEqual('/initial-path');
+      rerender(
+        <CommonTestWrapper>
+          <Router history={history}>
+            <NamespaceContext.Provider value='ns1'>
+              <EnhancedCompareV2 {...generateProps()} />
+            </NamespaceContext.Provider>
+          </Router>
+        </CommonTestWrapper>,
+      );
+      expect(history.location.pathname).toEqual('/initial-path');
+    });
+  });
```

Comment:
These tests were simply modeled after the existing tests in the `CompareV1.test.tsx` file: https://github.com/kubeflow/pipelines/blob/13736b85b6f11c5e75656aec5a427e0b1b0e9345/frontend/src/pages/CompareV1.test.tsx#L616-L682